### DISCUSSION TITLE
codepy NOT working with gcc 11.2.0

### DISCUSSION THREAD

**Discussion by Overcraft90** [inducer/codepy#33] [Troubleshooting]
Hi there,

I'm struggling to make `codepy` to work with the following `gcc version`:

> gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0
> Copyright (C) 2021 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Now, I really need it because it is required by a tool that performs scaffolding and leverages both GPU acceleration and PyCUDA. If it is helpful I cna report the error; however, based on what the developer for the tool said `codepy` for working need `gcc vesion` not greater than 8.x.x.

On the other hand, I really do not want to remove my `gcc version` since this might cause issues with XOrg driver (which I experienced before and are rather unpleasant to deal with). I also tried to install `gcc 8.5.0` from shell, but unsuccessfully; later, I resorted to install it from the _.bin file_ but even so, there is no way for me to use it in combination with the tool I need.

Thus, I was wondering is there any chance I can make it work with `gcc 11.2.0`

**Comment by inducer** [comment]
Your report doesn't nearly have enough information for anyone to help you.

- What did you do? (precisely)
- What happened?
- What did you expect to happen instead?

**Comment by Overcraft90** [comment]
Sorry, I apologize I will start explaining what I was doing.

I basically tried to scaffold a human genome using a tool ([instaGRAAL](https://github.com/koszullab/instaGRAAL)) which requires both CUDA and PyCUDA to be installed. I did so, and after struggling a bit, I got both my `nvidia-smi` & `nvcc -V` commands showing the correct output.

However, after few moments the tool was running I faced the issue reported in this log file, [stdout_codepy.log](https://github.com/inducer/codepy/files/8675903/stdout_codepy.log), which based on the developers of the tool is related to an incompatibility between `GCC` and `codepy`... 
Here, it is also mentioned **_toolchian.py_** which I'm not familiar with neither do the developers addressed that particular script.

Coming to what I'm expecting to get, I essentially should have the tool processing my scaffolds (portions of the human genome) so that they can be re-ordered according to the information about their positions generated by the tool in a former step, which didn't call for `codepy` yet.

Let me know if you need any further details.

**Comment by inducer** [comment]
I'd say your main problem is
```
/usr/bin/ld: cannot find -lpython3.10m: No such file or directory
```
Codepy seems to think that file should be in `/usr/lib/x86_64-linux-gnu`.

**Comment by Overcraft90** [comment]
Hi again,

You were right! Sorry but I'm quite new to the field and couldn't figure out the issue myself... However, as per another library, I created a soft-link to the one the tool was pointing at.

Now it works perfectly, thanks a lot! I think we can close this issue.
